### PR TITLE
[KOGITO-5947] BDD - Kogito Services failover tests

### PR DESCRIPTION
**Pull request by radtriste**
https://issues.redhat.com/browse/KOGITO-5947

Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [x] You have read the [contributors' guide](https://github.com/kiegroup/kogito-operator/blob/main/README.md#contributing-to-the-kogito-operator)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains a link to the JIRA issue
- [x] Pull Request contains a description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a unit test that verifies it
- [x] You've ran `make before-pr` and everything is working accordingly
- [ ] You've tested the new feature/bug fix in an actual OpenShift cluster
- [ ] You've added a [RELEASE_NOTES.md](https://github.com/kiegroup/kogito-operator/blob/main/RELEASE_NOTES.md) entry regarding this change

### CODE REVIEW

File: test/features/install_dataindex.feature [hunk 1]
```diff
@@ -89,4 +89,225 @@ Feature: Kogito Data Index
...
+      {
+        "approver" : "john",
+        "order" : {
+          "orderNumber" : "12345",
+          "shipped" : false
+        }
+      }
+      """
+    
+    Then Service "process-quarkus-example" contains 1 instances of process with name "orders"
```
Comment (sutaakar) [inline test/features/install_dataindex.feature:145]: So the service lost previous process? I would expect that it is still available as Infinispan use persistent volume.

Comment (radtriste) [reply test/features/install_dataindex.feature:145]: This is normal behavior as we use the cache service: https://infinispan.org/docs/infinispan-operator/main/operator.html#_data_grid_service
btw, we may need to change it to DataGrid (see the warning)

Comment (sutaakar) [reply test/features/install_dataindex.feature:145]: Right, we should migrate soon.
Fine for me then.

**Review by sutaakar — APPROVED** [review]

File: test/features/install_dataindex.feature (hunk 1 above)
Comment (radtriste) [reply test/features/install_dataindex.feature:145]: Created https://issues.redhat.com/browse/KOGITO-6237 and will try to do that next week

**Pull request merged**